### PR TITLE
Rebrand Terraform Cloud to HCP Terraform

### PR DESCRIPTION
**Pull request by sebasslash**
Rebrand Terraform Cloud to HCP Terraform

## Target Release

1.8.2

### ENHANCEMENTS 

 

-  Rebrands all references from "Terraform Cloud" to "HCP Terraform"

## More Notes
You should expect no functional changes regarding the rebrand:
- All CLI output that contains "Terraform Cloud", "TFC/E" has been changed to instead show "HCP Terraform" and/or "Terraform Enterprise". 
- The `cloud` block will not be renamed
- All `cloud` backend related env vars such as `TF_CLOUD_ORGANIZATION`, `TF_CLOUD_PROJECT`, etc are **not** being renamed.

### CODE REVIEW

**Review by Maed223 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ blank_issues_enabled: false
 contact_links:
   - name: HCP Terraform and Terraform Enterprise Troubleshooting and Feature Requests
     url: https://support.hashicorp.com/hc/en-us/requests/new
-    about: For issues and feature requests related to HCP Terraform and Terraform Enterprise, please submit a HashiCorp support request or email tf-cloud@hashicorp.support
+    about: For issues and feature requests related to the HCP Terraform or Terraform Enterprise, please submit a HashiCorp support request or email tf-cloud@hashicorp.support
```
Comment (crw) [inline .github/ISSUE_TEMPLATE/config.yml:8]: "the" HCP Terraform, is that correct usage?

File: internal/cloud/backend.go [hunk 2]
```diff
@@ -419,6 +430,19 @@ func (b *Cloud) Configure(obj cty.Value) tfdiags.Diagnostics {
...
 }
 
+func (b *Cloud) AppName() string {
+	if isValidAppName(b.appName) {
+		return b.appName
+	}
+	return "HCP Terraform"
+}
+
+// TODO: REMOVE WHEN go-tfe's IsCloud() returns true for
```
[the remark this answers is not in the document]
Comment (sebasslash) [reply internal/cloud/backend.go:440]: Yep, I will need rebranded go-tfe released first to fix this.

File: internal/command/meta_backend_migrate.go [hunk 3]
```diff
@@ -1003,21 +1004,22 @@ For example, if a workspace is currently named 'prod', the pattern 'app-*' would
...
+workspaces are named uniquely across all configurations used within an organization. A typical 
+strategy to start with is <COMPONENT>-<ENVIRONMENT>-<REGION> (e.g. networking-prod-us-east, 
+networking-staging-us-east).
 
 For more information on workspace naming, see https://www.terraform.io/docs/cloud/workspaces/naming.html
 
-When migrating existing workspaces from the backend %[1]q to Terraform Cloud, would you like to
-rename your workspaces? Enter 1 or 2.
+When migrating existing workspaces from the backend %[1]q to HCP Terraform or Terraform Enterprise, 
+would you like to rename your workspaces? Enter 1 or 2.
```
Comment (nfagerlund) [inline internal/command/meta_backend_migrate.go:1015]: Hmm, this instance of the "x or y" compromise is somewhat more frustrating than most of the others. 

We don't currently have the name available in the `promptMultiStateMigrationPattern` helper function, but that func's call site _does_ have access to a full cloud backend (in opts.Destination), so it ought to be possible to do a type-assert, grab the .AppName() off it, and pass it into the helper as a second arg. 

Reckon it's worth it? The approach would also be applicable to some of the other messages in this zone.

File: internal/command/show.go [hunk 4]
```diff
@@ -285,7 +285,7 @@ func (c *ShowCommand) getDataFromCloudPlan(plan *cloudplan.SavedPlanBookmark, re
 	// Cloud plans only work if we're cloud.
 	cl, ok := b.(*cloud.Cloud)
 	if !ok {
-		return nil, errUnusable(fmt.Errorf("can't show a saved cloud plan unless the current root module is connected to Terraform Cloud"), "cloud plan")
+		return nil, errUnusable(fmt.Errorf("can't show a saved cloud plan unless the current root module is connected to HCP Terraform"), "cloud plan")
```
Comment (nfagerlund) [inline internal/command/show.go:288]: I think this one should get the "or", alas.

File: internal/command/test.go [hunk 5]
```diff
@@ -42,12 +42,12 @@ Usage: terraform [global options] test [options]
 
 Options:
 
-  -cloud-run=source     If specified, Terraform will execute this test run
-                        remotely using Terraform Cloud. You must specify the
+  -cloud-run=source     If specified, Terraform will execute this test run 
+                        remotely using HCP Terraform. You must specify the
```
Comment (nfagerlund) [inline internal/command/test.go:46]: Should this one get the "or" also?

**Review by nfagerlund — COMMENTED** [review]
Question: should this PR wait for a release of the rebranded go-tfe? 

This mostly seems to be in good shape. There's some amount of inconsistency around when we do or don't use the "cloud or enterprise" construction; this PR doesn't exactly add more of it, but I should ask — how much of it do you think is worth addressing at this time? We can hammer out a rubric for when to do it and when not to, and I can go through with a fine-toothed comb to enforce it... but I wanna let you be the judge of whether it's worth it, because you know more about the marketing timelines for this than we do.

File: internal/command/meta_backend_migrate.go [hunk 6]
```diff
@@ -565,23 +565,23 @@ func (m *Meta) backendMigrateTFC(opts *backendMigrateOpts) error {
...
-		// From Terraform Cloud to another backend. This is not yet implemented, and
-		// we recommend people to use the TFC API.
+		// From HCP Terraform to another backend. This is not yet implemented, and
+		// we recommend people to use the HCP Terraform API.
 		return fmt.Errorf(strings.TrimSpace(errTFCMigrateNotYetImplemented))
 	}
 
 	// Everything below, by the above two conditionals, now assumes that the
-	// destination is always Terraform Cloud (TFC).
+	// destination is always HCP Terraform (TFC).
```
Comment (BrianMMcClain) [inline internal/command/meta_backend_migrate.go:576]:
```suggestion
	// destination is always HCP Terraform.
```

File: internal/rpcapi/cli.go [hunk 7]
```diff
@@ -78,10 +78,10 @@ func (c cliCommand) Run(args []string) int {
 			// TODO:
 			//
 			// The following message says that this interface is for Terraform
-			// Cloud only because we're using Terraform Cloud's integration
+			// Cloud only because we're using HCP Terraform's integration
```
Comment (BrianMMcClain) [inline internal/rpcapi/cli.go:81]: Flagging this, but a reference to "Terraform Cloud" remains split across lines 80 and 81

**Review by BrianMMcClain — APPROVED** [review]

**Review by BrianMMcClain — APPROVED** [review]

File: internal/cloud/testdata/plan-json-full/plan-redacted.json [hunk 8]
```diff
@@ -1 +1 @@
...pe":{"attributes":{"bool":{"type":"bool","description":"An optional boolean attribute, can be true or false.","description_kind":"markdown","optional":true},"float":{"type":"number","description":"An optional float attribute.","description_kind":"markdown","optional":true},"integer":{"type":"number","description":"An optional integer attribute.","description_kind":"markdown","optional":true},"number":{"type":"number","description":"An optional number attribute, can be an integer or a float.","description_kind":"markdown","optional":true},"string":{"type":"string","description":"An optional string attribute.","description_kind":"markdown","optional":true}},"nesting_mode":"set"},"description":"A set attribute that contains objects that match the root schema, allowing for nested collections and objects to be modelled.","description_kind":"markdown","optional":true},"string":{"type":"string","description":"An optional string attribute.","description_kind":"markdown","optional":true}},"nesting_mode":"map"},"description":"A map attribute that contains objects that match the root schema, allowing for nested collections and objects to be modelled.","description_kind":"markdown","optional":
```
[the remark this answers is not in the document]
Comment (sebasslash) [reply internal/cloud/testdata/plan-json-full/plan-redacted.json:1]: That's okay, we'll leave it as is given this is just a test fixture.

**Review by nfagerlund — APPROVED** [review]
All right, this looks good enough to work with! The "HCP or TFE" curse is as frustrating as ever when it comes to making all this text work sensibly, but your mix of approaches and fallbacks here seems roughly as good as one could hope for. 👍🏼

File: internal/command/meta_backend_migrate.go (hunk 6 above)
Comment (Yuksell06) [reply internal/command/meta_backend_migrate.go]: []()

**Review by Yuksell06 — CHANGES_REQUESTED** [review]
Herşey çok daha güzel olacak

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yuksell06) [reply .github/ISSUE_TEMPLATE/config.yml]: Enes

**Pull request merged**